app/questionnaire/questionnaire_manager.py

```diff
@@ -43,7 +42,7 @@ def validate_all_answers(self):
             is_valid = self.validate(location, answers)
 
             if not is_valid:
-                logger.debug("Failed validation with current location %s", str(location))
+                logger.debug("failed validation", **location.__dict__)
```

Comment:
Might be better to define "__str__" on the location rather than calling this internal method?